web/libs/datamanager/src/components/MainView/GridView/GridView.jsx

```diff
@@ -215,6 +232,70 @@ export const GridView = observer(({ data, view, loadMore, fields, onChange, hidd
     [data.length, view.dataStore.hasNextPage],
   );
 
+  // Handle column count changes
+  useEffect(() => {
+    const prevColumnCount = prevColumnCountRef.current;
+    const currentColumnCount = columnCount;
+
+    // If column count changed and we have more columns now (showing fewer rows)
+    if (prevColumnCount !== currentColumnCount) {
+      prevColumnCountRef.current = currentColumnCount;
+
+      // Calculate how many items we can display with the new column count
+      const estimatedVisibleRows = Math.ceil(window.innerHeight / finalRowHeight);
+      const estimatedVisibleItems = estimatedVisibleRows * currentColumnCount;
+
+      // If we don't have enough items to fill the visible area, load more
+      // Note: We don't check !view.dataStore.loading here because we want to trigger loading
+      // even if something is already loading, to ensure we get enough items
+      if (data.length < estimatedVisibleItems && view.dataStore.hasNextPage) {
+        loadMore?.();
+      }
+
+      // Fallback: if we have significantly fewer items than columns, always load more
+      if (data.length < currentColumnCount * 2 && view.dataStore.hasNextPage) {
+        loadMore?.();
+      }
+
+      // Special case: if we have fewer items than the column count itself, definitely load more
+      // This handles the case where there aren't enough items to even fill one row
+      if (data.length < currentColumnCount && view.dataStore.hasNextPage) {
+        loadMore?.();
+      }
+    }
+  }, [columnCount, data.length, view.dataStore.hasNextPage, view.dataStore.loading, loadMore, finalRowHeight]);
```

Comment:
```suggestion
  }, [columnCount, data.length, view.dataStore.hasNextPage, loadMore, finalRowHeight]);
```
Do we want to be reacting to loading, your mention of:

> Note: We don't check !view.dataStore.loading here because we want to trigger loading

Makes me wonder if this could runaway?